pysidt/sidt.py

```diff
@@ -548,20 +548,20 @@ def generate_tree(
         self,
         data=None,
         check_data=True,
-        test=None,
+        val=None,
```

Comment:
Yes, test is a bad name for this, but is `val` a good name for it?

Reply:
If `validation` better? `val` can be confused with value?